test/functional/shared/cli/interactive_test.go

```diff
@@ -0,0 +1,133 @@
+	"testing"
+
+	expect "github.com/Netflix/go-expect"
+	"github.com/creack/pty"
+	"github.com/hinshun/vt10x"
+	"github.com/stretchr/testify/require"
+)
+
+// Prompt provides test utility for prompt test.
+type Prompt struct {
+	CmdArgs   []string
+	Procedure func(*testing.T, *expect.Console) error
+}
+
+type InteractiveCommandRunner struct {
+}
+
+func NewInteractiveCommandRunner() *InteractiveCommandRunner {
+	return &InteractiveCommandRunner{}
+}
+
+func (r *InteractiveCommandRunner) Run(t *testing.T, prompt Prompt) {
+	t.Helper()
+
+	// Prepare the pseudo-terminal.
+	//
+	// pty is the main part of the pseudo-terminal.
+	// > It is what your application will interact with when it wants to control the terminal.
+	// > Reading from pty gets the output of the terminal, and writing to pty writes input to the terminal.
+	// tty is the other end of the terminal.
+	// > It is what you would see on the screen in a terminal application.
+	pty, tty, err := pty.Open()
+	require.NoError(t, err)
+	defer func() { _ = pty.Close() }() // Close pty when done.
+	defer func() { _ = tty.Close() }() // Close tty when done.
+
+	// Create a terminal emulator.
+	term := vt10x.New(vt10x.WithWriter(tty))
+
+	// Create a console.
+	c, err := expect.NewConsole(
+		expect.WithStdin(pty),
+		expect.WithStdout(term),
+		expect.WithCloser(pty, tty),
+	)
+	require.NoError(t, err)
+	defer c.Close() // Close console when done.
+
+	// Prepare and start command on terminal emulator
+	cmd := exec.Command("rad", prompt.CmdArgs...)
+	cmd.Stdin = tty
+	cmd.Stdout = tty
+	cmd.Stderr = tty
+
+	go func() {
+		if err := prompt.Procedure(t, c); err != nil {
+			t.Errorf("procedure failed: %v", err)
+		}
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		log.Fatalf("cmd.Start() = %v", err)
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		log.Fatalf("cmd.Wait() = %v", err)
+	}
+}
+
+func TestInteractiveRadiusCommands(t *testing.T) {
+	// t.Parallel()
+
+	runner := NewInteractiveCommandRunner()
+
+	t.Run("rad init with application to be scaffolded", func(t *testing.T) {
+		runner.Run(t, Prompt{
+			CmdArgs: []string{"init"},
+			Procedure: func(t *testing.T, c *expect.Console) error {
+				_, err := c.ExpectString("Setup application in the current directory?")
+				require.NoError(t, err)
+
+				// Trying to send Enter key press.
+				_, err = c.SendLine("")
+				require.NoError(t, err)
+
+				return nil
+			},
+		})
+	})
+
+	t.Run("rad init with application not to be scaffolded", func(t *testing.T) {
+		runner.Run(t, Prompt{
+			CmdArgs: []string{"init"}, // Assuming that "--yes" is a valid argument for "rad init"
+			Procedure: func(t *testing.T, c *expect.Console) error {
+				_, err := c.ExpectString("Setup application in the current directory?")
+				require.NoError(t, err)
+
+				// `j` is equal to CursorDown.
+				_, err = c.Send("j")
+				require.NoError(t, err)
+
+				// Trying to send Enter key press.
+				_, err = c.SendLine("")
+				require.NoError(t, err)
+
+				return nil
+			},
```

Comment:
What does a failure look like when:

- The console output doesn't match expected?
- If the CLI reports an error?
- If the test gets stuck and hits a timeout? (eg: lets say we change the "down" shortcut" to something other than `j`)

Reply:
As an example of why I'm asking .... we have had issues with local-dev recipes since migrating to GHCR. 

Let's say that pulling/installing from GHCR works 99% of the time, how would we use this test to understand that failure?